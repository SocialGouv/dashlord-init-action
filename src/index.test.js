const fs = require("fs");
const core = require("@actions/core");

jest.mock("fs");

const { getOutputs } = require("./index");

const sampleConfig = `
title: Test 1
urls:
  - url: https://www.free.fr
    title: Free
    repositories:
      - iliad/free-ui
      - iliad/free-api
  - url: invalid-url
  - url: http://chez.com
    repositories:
      - chez/chez-ui
      - chez/chez-api
  - url: https://voila.fr
`;

let inputs = {};

describe("should parse dashlord config", () => {
  beforeAll(() => {
    // Mock getInput
    jest.spyOn(core, "getInput").mockImplementation((name) => {
      return inputs[name];
    });
  });
  beforeEach(() => {
    // Reset inputs
    inputs = {};
  });
  test("when no input", async () => {
    fs.existsSync.mockReturnValue(true);
    fs.readFileSync.mockReturnValue(sampleConfig);
    const outputs = getOutputs();
    expect(outputs.urls_json).toMatchSnapshot();
  });

  test("when single invalid url input", async () => {
    inputs.url = 'zfzef'
    fs.existsSync.mockReturnValue(true);
    fs.readFileSync.mockReturnValue(sampleConfig);
    const outputs = getOutputs();
    expect(outputs.urls_json).toMatchSnapshot();
  });

  test("when single valid url input", async () => {
    inputs.url = 'https://www.free.fr'
    fs.existsSync.mockReturnValue(true);
    fs.readFileSync.mockReturnValue(sampleConfig);
    const outputs = getOutputs();
    expect(outputs.urls_json).toMatchSnapshot();
  });

    test("when multiple urls input", async () => {
    inputs.url = 'https://www.free.fr,pouet,http://chez.com';
    fs.existsSync.mockReturnValue(true);
    fs.readFileSync.mockReturnValue(sampleConfig);
    const outputs = getOutputs();
    expect(outputs.urls_json).toMatchSnapshot();
  });

});
