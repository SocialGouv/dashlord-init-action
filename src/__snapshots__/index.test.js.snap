// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should parse dashlord config when multiple urls input 1`] = `
Array [
  Object {
    "repositories": Array [
      "iliad/free-ui",
      "iliad/free-api",
    ],
    "title": "Free",
    "url": "https://www.free.fr",
  },
  Object {
    "repositories": Array [
      "chez/chez-ui",
      "chez/chez-api",
    ],
    "url": "http://chez.com",
  },
]
`;

exports[`should parse dashlord config when no input 1`] = `
Array [
  Object {
    "repositories": Array [
      "iliad/free-ui",
      "iliad/free-api",
    ],
    "title": "Free",
    "url": "https://www.free.fr",
  },
  Object {
    "repositories": Array [
      "chez/chez-ui",
      "chez/chez-api",
    ],
    "url": "http://chez.com",
  },
  Object {
    "url": "https://voila.fr",
  },
]
`;

exports[`should parse dashlord config when single invalid url input 1`] = `Array []`;

exports[`should parse dashlord config when single valid url input 1`] = `
Array [
  Object {
    "repositories": Array [
      "iliad/free-ui",
      "iliad/free-api",
    ],
    "title": "Free",
    "url": "https://www.free.fr",
  },
]
`;
